common/src/types.rs

```diff
@@ -143,3 +146,28 @@ pub enum SubscriptionEvent<N: NetworkSpec> {
 }
 
 pub type SubEventRx<N> = Receiver<SubscriptionEvent<N>>;
+
+// TODO: Might have to generalise this for other VM
+#[derive(Debug, Error)]
+pub enum EvmError {
+    #[error("execution reverted: {}", display_revert(.0))]
+    Revert(Option<Bytes>),
+
+    #[error("evm error: {0:?}")]
+    Generic(String),
+
+    #[error("rpc error: {0:?}")]
+    RpcError(Report),
+}
+
+fn display_revert(output: &Option<Bytes>) -> String {
+    match output {
+        Some(bytes) => decode_revert_reason(bytes.as_ref()).unwrap_or(hex::encode(bytes)),
+        None => "execution halted".to_string(),
+    }
+}
+
+pub struct AccessListResultWithAccounts {
```

Comment:
Doesn't this contain redundant info? The access list result contains accounts already?

Reply:
AccessListResult does contain a vector of accounts but it is not in form of a `HashMap<address,account>`, it'd need some level of parsing to retrieve that format and thus I think it might have been kept for `create_extended_access_list` . Used here
https://github.com/a16z/helios/blob/e030480100ff2b21a218eda076a9a80b2f55978c/verifiable-api/server/src/service.rs#L195

Altho the other impl goes on with parsing route on rpc https://github.com/a16z/helios/blob/e030480100ff2b21a218eda076a9a80b2f55978c/core/src/execution/client/rpc.rs#L138

Can keep whatever you feel is correct